common/network/components/responder.c

```diff
@@ -6,21 +6,104 @@
  */
 
 #include "common/network/components/responder.h"
+#include "ciri/core.h"
+#include "ciri/node.h"
+#include "common/storage/sql/defs.h"
+#include "common/storage/storage.h"
 #include "utils/logger_helper.h"
 
 #define RESPONDER_COMPONENT_LOGGER_ID "responder_component"
 
+static bool get_transaction_to_reply(responder_state_t *const state,
+                                     hash_request_t *const request,
+                                     iota_transaction_t *const tx) {
+  if (state == NULL) {
+    return false;
+  }
+  if (request == NULL) {
+    return false;
+  }
+  if (request->neighbor == NULL) {
```

Comment:
what if the neighbour was removed in between the request being added to the list & being responded to

Reply:
Right, I think we need to copy that neighbor (at least the endpoint part) and let the sending fail